packages/ckeditor5-find-and-replace/src/findcommand.ts

```diff
@@ -60,16 +60,25 @@ export default class FindCommand extends Command {
 		const findAndReplaceUtils: FindAndReplaceUtils = editor.plugins.get( 'FindAndReplaceUtils' );
 
 		let findCallback: FindCallback | undefined;
+		let callbackSearchText: string = '';
 
 		// Allow to execute `find()` on a plugin with a keyword only.
 		if ( typeof callbackOrText === 'string' ) {
-			findCallback = findAndReplaceUtils.findByTextCallback( callbackOrText, { matchCase, wholeWords } );
-
-			this._state.searchText = callbackOrText;
+			findCallback = ( ...args ) => ( {
+				results: findAndReplaceUtils.findByTextCallback( callbackOrText, { matchCase, wholeWords } )( ...args ),
+				searchText: callbackOrText
+			} );
 		} else {
 			findCallback = callbackOrText;
 		}
 
+		// Set the search text to the last search text if it was not provided.
+		findCallback = tapFunctionResult( findCallback, findCallbackResult => {
+			if ( findCallbackResult && !Array.isArray( findCallbackResult ) ) {
+				callbackSearchText = findCallbackResult.searchText;
+			}
+		} );
```

Comment:
I would avoid such generic wrapper here (especially that it's not used anywhere else), maybe there is a simple, more straightforward way to do this?

Reply:
Generic wrapper removed.